dcrec/secp256k1/sigerror_test.go

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2020 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package secp256k1
+
+import (
+	"errors"
+	"testing"
+)
+
+// TestSignatureErrorCodeStringer tests the stringized output for the
+// SignatureErrorCode type.
+func TestSignatureErrorCodeStringer(t *testing.T) {
```

Comment:
this test seems a bit over the top.. I'm sure there's a better way.

Related, but I also expect that the `stringer` utility will cut down on code size, because there does not need to be any init code to build the map in that implementation, and would reduce time spent scanning the heap during garbage collection because there are fewer pointers.  Going to do some experimenting on that idea.